src/components/Layout/FormPageLayout.tsx

```diff
@@ -0,0 +1,175 @@
+import { ObjectState } from "@homebound/form-state";
+import { Observer } from "mobx-react";
+import { Css } from "src/Css";
+import { BoundForm, BoundFormInputConfig } from "src/forms";
+import { useTestIds } from "src/utils";
+import { Button, ButtonProps } from "../Button";
+import { Icon, IconKey } from "../Icon";
+import { HeaderBreadcrumb, PageHeaderBreadcrumbs } from "./PageHeaderBreadcrumbs";
+
+export type FormSectionConfig<F> = {
+  title?: string;
+  icon?: IconKey;
+  rows: BoundFormInputConfig<F>;
+}[];
+
+type ActionButtonProps = Pick<ButtonProps, "onClick" | "label" | "disabled" | "tooltip">;
+
+type FormPageLayoutProps<F> = {
+  pageTitle: string;
+  breadCrumb?: HeaderBreadcrumb | HeaderBreadcrumb[];
+  formState: ObjectState<F>;
+  formSections: FormSectionConfig<F>;
+  submitAction?: ActionButtonProps;
+  cancelAction?: ActionButtonProps;
+  tertiaryAction?: ActionButtonProps;
+};
+
+/** In order to make the multiple stacked sticky elements work (Header, then sidebar below) we need to set the header height.
+ * The alternate solution was to have multiple defined grid areas for the fixed header vs body below, but this becomes more complex
+ * to manage when adding in a max content width container for the page while keeping the page scrollbar to the far right of the page.
+ * Rather than wrapping the page in a max-width div, we use "gutter" columns `minMax(0, auto)` that kick in when all other columns have met their max widths.
+ */
+const headerHeightPx = 120;
+
+export function FormPageLayout<F>(props: FormPageLayoutProps<F>) {
+  const { formSections, formState } = props;
+
+  const tids = useTestIds(props, "formPageLayout");
+
+  const gridColumns =
+    "minMax(0, auto) minMax(100px, 250px) minMax(350px, 1000px) minMax(min-content, 300px) minMax(0, auto)";
+
+  return (
+    // This page is `fixed` to the full screen to allow it to act as a full screen modal while content is mounted below
```

Comment:
"while content is mounted below"? can you clarify what other content this would be? is this like a table with 100 rows, and so opening this "page" to edit row 1, doesn't throw away the table itself/loosing your scroll position/etc?

Reply:
yeah exactly, will add more context to the comment